NetworkConfigurations/NetworkAddressingAndFilteringSubCategories/DHCP.md

```diff
@@ -0,0 +1,263 @@
+
+```
+systemctl status isc-dhcp-server.service
+● isc-dhcp-server.service - LSB: DHCP server
+   Loaded: loaded (/etc/init.d/isc-dhcp-server; generated; vendor preset: enable
+   Active: failed (Result: exit-code); 10s ago
+```
+
+You may observe that DHCP isn't configured yet, and it displays a failed status. To rectify this, we need to adjust the DHCP server configuration.
+
+To accomplish this, access the configuration file located at `/etc/dhcp/dhcpd.conf` and make the necessary changes. Use the following command:
+
+```
+nano /etc/dhcp/dhcpd.conf
+```
+
+Within the file, modify the contents as required. Below is a basic sample configuration:
+
+```
+subnet 192.168.1.0 netmask 255.255.255.0 {
+    range 192.168.1.100 192.168.1.200;
+    option routers 192.168.122.1;
+    option domain-name-servers 192.168.122.1;
+    default-lease-time 86400;
+    max-lease-time 86400;
+}
+```
+
+You also may need to edit `/etc/default/isc-dhcp-server` to specify the interfaces DHCP should listen to.
+
+```
+nano /etc/default/isc-dhcp-server
+```
+
+Within the file, specify the interfaces.
+
+```
+INTERFACESv4="br0"
+```
+
+After modifying the configuration, if you attempt to restart the DHCP server, you may encounter a failure. To restart, use the following command:
+
+```
+sudo systemctl restart isc-dhcp-server
+Job for isc-dhcp-server.service failed because the control process exited with error code.
+See "systemctl status isc-dhcp-server.service" and "journalctl -xe" for details.
+```
+
+This issue arises because the IP address assigned to the interface of DENT needs to be appropriate for DHCP server usage. For instance, if a bridge interface named br0 exists, it must have an IP address from which other devices can be assigned IPs within the DHCP server range.
+
+To assign an IP address to br0, execute the following command:
+
+```
+ip address add 192.168.1.10/24 dev br0
+```
+
+Afterward, restart the DHCP server using `sudo systemctl restart isc-dhcp-server`. This time, it should operate without errors.
+
+Additionally, verify the status to ensure it's active and running:
+
+```
+systemctl status isc-dhcp-server.service
+● isc-dhcp-server.service - LSB: DHCP server
+   Loaded: loaded (/etc/init.d/isc-dhcp-server; generated; vendor preset: enable
+   Active: active (running)
+```
+
+Now, the DHCP server is ready to randomly assign IP addresses to different devices from the specified range.
+
+### Example:
+
+**Using DHCP with a Bridge:**
+
+1. Begin by creating a bridge interface.
+
+   ```
+   ip link add name br0 type bridge
+   ```
+
+2. Set the desired ports (`enp0s4` and `enp0s5`) as members of the bridge.
+
+   ```
+   ip link set enp0s4 master br0
+   ip link set enp0s5 master br0
+   ```
+
+3. Bring the interface 'br0' and the ports `enp0s4` and `enp0s5` up.
+
+   ```
+   ip link set br0 up
+   ip link set enp0s4 up
+   ip link set enp0s5 up
+   ```
+
+4. Initiate the DHCP client on the bridge interface (`br0`).
```

Comment:
why do you need to get IP on br0 with DHCP? usually It has to be static

Reply:
If you want me to remove this, I'll, but if we refer to the section 'To start a DHCP client on a bridge' [here](https://github.com/Marvell-switching/switchdev-prestera/wiki/DHCP), it states the same procedure for a bridge to act as a client. However, I have also ensured to add a static IP- https://github.com/dentproject/docs.dent.dev/pull/28/files#diff-3fe5243907877b6184ff15c1f302b39a8a39018c644e1c89161efa29ec25901cR92.

Reply:
I'd recommend splitting this document into 2 sections. 1) DHCP client 2) DHCP server.

when DHCP server is configured to listen on br0 there Is no point in getting IP address for br0 from that DHCP server, same for vlan sub-interfases

Reply:
Or can we just have this example without the last step of getting the IP address for br0, since we don't require it?